stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/utils/extensions/Channel.kt

```diff
@@ -1,11 +1,8 @@
 package com.getstream.sdk.chat.utils.extensions
 
-import io.getstream.chat.android.client.extensions.getUsers
+import io.getstream.chat.android.client.extensions.getUsersExcludingCurrent
 import io.getstream.chat.android.client.models.Channel
 import io.getstream.chat.android.core.internal.InternalStreamChatApi
-import io.getstream.chat.android.offline.ChatDomain
 
 @InternalStreamChatApi
-public fun Channel.isDirectMessaging(currentUserId: String = currentUserId()): Boolean = getUsers(currentUserId).size == 1
-
-private fun currentUserId(): String = ChatDomain.instance().user.value?.id ?: ""
+public fun Channel.isDirectMessaging(): Boolean = getUsersExcludingCurrent().size == 1
```

Comment:
It doesn't seem very efficient to create this list of users with a bunch of operations to just check for its size. How about

```suggestion
public fun Channel.isDirectMessaging(): Boolean {
    return members.size == 2 && includesCurrentUser()
}

private fun Channel.includesCurrentUser(): Boolean {
    val currentUserId = ChatClient.instance().getCurrentUser()?.id ?: return false
    return members.any { it.user.id == currentUserId }
}
```